packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -161,7 +162,10 @@ export const useGridKeyboardNavigation = (
     GridEventListener<GridEvents.columnHeaderNavigationKeyDown>
   >(
     (params, event) => {
-      event.preventDefault();
+      const isNativeFocusOnHeader = event.currentTarget === event.target;
+      if (isNavigationKey(event.key) && isNativeFocusOnHeader) {
+        event.preventDefault();
```

Comment:
If the key is not a navigation key shouldn't it do nothing? If the focus is in another element of the column header cell, lets say the sort button, it won't call `event.preventDefault()` but it still tries to navigate to the cell below.

_Click into the sort icon button then press <kbd>Arrow Down</kbd>_

<img width="400" src="https://user-images.githubusercontent.com/42154031/149608289-65cf5fe1-969a-42e9-8412-772a34fabf78.gif" alt="msedge_3YcFi2d6YP" style="max-width: 100%;">

Reply:
It tooks me time to understand what append

In the PR #3275 that was only refact for column header key down, I missed the following condition
```js
if (!isGridHeaderCellRoot(event.target as HTMLElement)) {
        return;
}
```
https://github.com/mui-org/material-ui-x/pull/3275/files#diff-130975bb0d3117df317939124519a996a0f43f29799b345003e063ea6c1a181aL103-L124

Before the introduction of the bug, the rule was the next one: If the focus is not on the root component, the `columnHeaderKeyDown` event has no effect. One exception is for that checkbox selection which has the focus set in the checkbox input.

I think this PR is a "regression" fix and I will open an issue to discuss how events from custom render should be managed.

Reply:
Note that `isGridHeaderCellRoot` only checks if `event.target` is the column header. If the target is a child of the header (e.g. the sort icon) it evaluates to false. This creates a regression because if I click the sort icon and I press <kbd>Arrow Down</kbd> it will scroll the page instead of going to the cell below.

> I think this PR is a "regression" fix and I will open an issue to discuss how events from custom render should be managed.

Does this mean that you won't be fixing the column header with a custom component? If yes, could you update the title and description so #3599 doesn't get closed once this one is merged

Reply:
Yes, `isGridHeaderCellRoot` is a hard condition, but that is what we used before the regression occurs. 

You can test the codesandbox from the issue with different versions of DataGrid:
https://codesandbox.io/s/renderheadergrid-material-demo-forked-suwsf?file=/index.tsx

- v5.2.0: 
	- TextField works perfectly
	- keyboard navigation is blocked if the focus is on the sort button
- v5.2.1: 
	- TextField broken
	- keyboard navigation is working if the focus is on the sort button

The modification between v5.2.0 and v5.2.1 is not intentional. So I propose to correct the regression I introduced in v5.2.1 to solve the issue.

With the modification added in the [commit](https://github.com/mui-org/material-ui-x/pull/3624/commits/8581bc248942b28839367f0d195735153a9ed58e) the <kbd>Arrow Down</kbd> keep same behavior as v5.2.0

Reply:
Why not to also return here? We only handle the navigation keys.

```suggestion
        event.preventDefault();
        return;
```

---

> keyboard navigation is working if the focus is on the sort button

I'm not sure about this statement. In https://deploy-preview-3624--material-ui-x.netlify.app/components/data-grid/demo/, if I click the sort icon, then press <kbd>Page Down</kbd>, it moves the page down.

Reply:
I could put an early return as follow

```js
if (isNavigationKey(event.key)) {
	event.preventDefault();
} else{
	return;
}
```
The page moves down, but not the grid. That's the behavior before the regression. Here is a comparison of key board before/after the regression and the preview of this commit.

From left to righ:
- before regression: https://deploy-preview-3302--material-ui-x.netlify.app/components/data-grid/demo/
- after regresion: https://deploy-preview-3275--material-ui-x.netlify.app/components/data-grid/demo/
- this PR: https://deploy-preview-3624--material-ui-x.netlify.app/components/data-grid/demo/

## When focus on sort
![Peek 2022-01-27 12-26](https://user-images.githubusercontent.com/45398769/151350055-82be1cbc-1feb-4c8c-9ff1-1bfe9f5d2c4c.gif)

## When focus on column header
![Peek 2022-01-27 12-27](https://user-images.githubusercontent.com/45398769/151350197-79549bf5-0d77-499f-978e-f5c0e9d8b8b6.gif)

Reply:
> I could put an early return as follow

The problem is `isGridHeaderCellRoot` only checking if the event was fired by the root.

Let's review what happened here:

- Before 5.2.1: clicking in any child of the column header and <kbd>ArrowDown</kbd> would scroll the page (nobody complained but I think it's not good)
- 5.2.1: Unintentionally fixed the bug above
- This PR reverts the fix

I see this as inserting a regression because 5.2.1 is already released. Could you take a look in #3692? I think it already fixed part of the problem here. The solution was to only call `event.preventDefault` for navigation keys. #3599 was kinda fixed for free by #3692. However, pressing a navigation key inside the input won't work.

Reply:
Yes, I had a look at #3692 It is equivalent to add
```js
if(isNavigationKey(event.key)) {
	event.preventDefault()
}
```

That was my first idea but I stopped doing it because it does not allow you to use keyboard navigation in a custom input, event more problematic, it does not allow you to use space key. The worst I imagine is putting a select input in the header. In such a case, the keyboard navigation will be completely down, because it requires keyboard arrows.

My conclusion is that only the developer knows if the focusable element added requires or not the `event.preventDefault` to prevent the page to scroll. But how to adapt the interface to let them choose, I don't know

Reply:
What about then wrapping the custom component in a div so we know from where the event is coming from and ignore it? I proposed this in https://github.com/mui-org/material-ui-x/pull/3624#pullrequestreview-853510614. I don't see a simpler and future-proof alternative. The other way, I think, would be to add a specific attribute to each of the icons and the title to allows us to detect if the event came from them or from a custom component.